### PR TITLE
(GH-9357) Clarify SessionOption for Invoke-Command

### LINKED ISSUE #9357: `Invoke-Command`: Example for SessionOption doesn't work
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Typo in `Get-Foo` cmdlet" instead of "Typo."
- [X] **Verify Version:** If there is a mismatch between documentation and the behavior on your system, ensure that the version you are using is the same as the documentation. Check this box if they match or the issue you are reporting is not version specific.

### Links

https://learn.microsoft.com/powershell/module/microsoft.powershell.core/invoke-command#example-18-start-scripts-on-many-remote-computers

### Summary

Example 18 on the Invoke Command page is not valid.

```powershell
$parameters = @{
  ComputerName = (Get-Content -Path C:\Test\Servers.txt)
  InDisconnectedSession = $true
  FilePath = "\\Scripts\Public\ConfigInventory.ps1"
  ------>SessionOption = @{OutputBufferingMode="Drop";IdleTimeout=43200000}
}
Invoke-Command @parameters
```

Tested on PowerShell 5.1 & 7.2.5 for Windows 10.

The command does not accept the hashtable. You can see related correspondence on this StackOverflow post https://stackoverflow.com/questions/57930750/invoke-command-indisconnectedsession-sessionoption-and-idletimeout-one-liner

### Details

Error thrown:
```
    $parameters = @{
        ComputerName = $foo
        InDisconnectedSession = $true
        Scriptblock = $somescriptblock
        ThrottleLimit = 50
        SessionOption = @{OutputBufferingMode="Drop";IdleTimeout=300000}
    }
    Invoke-Command @parameters -verbose

Invoke-Command : The specified IdleTimeout session option 0 (seconds) is not a valid period.  Specify an IdleTimeout value that is greater than or equal to the minimum allowed 60 (seconds).
At C:\Users\....\somescript.ps1:85 char:5
+     Invoke-Command @parameters -verbose
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Invoke-Command], PSArgumentException
    + FullyQualifiedErrorId : Argument,Microsoft.PowerShell.Commands.InvokeCommandCommand
```

However, running this worked:
```
Invoke-Command -SessionOption (New-PSSessionOption -OutputBufferingMode "Drop" -IdleTimeout 300000)
```

### Suggested Fix

Remove example 18 from the documentation or clarify usage of a Hash Table for the `-SessionOption` parameter.

### PR DESCRIPTION
**Pull request by michaeltlombardi**
# PR Summary

Prior to this change, Example 18 for `Invoke-Command` raised an error because the **IdleTimeout** value wasn't high enough. The root of this problem is that the example was casting a value for milliseconds to ticks. The **SessionOption** parameter documentation also wasn't explicit that specifying a hashtable for the parameter is different from calling `New-PSSessionObject`.

This change:

- Updates example 18 to use `[timespan]::FromHours()` to make the example functional and easier to understand
- Adds a note to the **SessionOption** parameter documentation about the behavior when a user specifies a hashtable
- Fixes Vale and markdownlint warnings where possible
- Standardizes example code snippets for readability
- Fixes #9357
- Fixes [AB#26713](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/26713)

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/Invoke-Command.md [hunk 1]
```diff
@@ -1222,6 +1256,15 @@ Specifies advanced options for the session. Enter a **SessionOption** object, su
...
 names and the values are session option values.
 
+> [!NOTE]
+> If you specify a hashtable for **SessionOption**, PowerShell converts the hashtable into a
+> **System.Management.Autiomation.Remoting.PSSessionOption** object. The values for keys specified
+> in the hashtable are cast to the matching property of the object. This behaves differently from
+> calling `New-PSSessionOption`. For example, the **System.TimeSpan** values for the timeout
+> properties, like **IdleTimeout**, convert an integer value into ticks instead of milliseconds.
+> For more information on the **PSSessionOption** object and its properties, see
+> [PSSessionOption](https://learn.microsoft.com/dotnet/api/system.management.automation.remoting.pssessionoption)
```
Comment (sdwheeler) [inline reference/5.1/Microsoft.PowerShell.Core/Invoke-Command.md:1266]:
```suggestion
> [PSSessionOption](/dotnet/api/system.management.automation.remoting.pssessionoption)
```

File: reference/7.0/Microsoft.PowerShell.Core/Invoke-Command.md [hunk 2]
```diff
@@ -1359,6 +1407,15 @@ Specifies advanced options for the session. Enter a **SessionOption** object, su
...
 names and the values are session option values.
 
+> [!NOTE]
+> If you specify a hashtable for **SessionOption**, PowerShell converts the hashtable into a
+> **System.Management.Autiomation.Remoting.PSSessionOption** object. The values for keys specified
+> in the hashtable are cast to the matching property of the object. This behaves differently from
+> calling `New-PSSessionOption`. For example, the **System.TimeSpan** values for the timeout
+> properties, like **IdleTimeout**, convert an integer value into ticks instead of milliseconds.
+> For more information on the **PSSessionOption** object and its properties, see
+> [PSSessionOption](https://learn.microsoft.com/dotnet/api/system.management.automation.remoting.pssessionoption)
```
Comment (sdwheeler) [inline reference/7.0/Microsoft.PowerShell.Core/Invoke-Command.md:1417]:
```suggestion
> [PSSessionOption](/dotnet/api/system.management.automation.remoting.pssessionoption)
```

File: reference/7.2/Microsoft.PowerShell.Core/Invoke-Command.md [hunk 3]
```diff
@@ -1378,6 +1426,15 @@ Specifies advanced options for the session. Enter a **SessionOption** object, su
...
 names and the values are session option values.
 
+> [!NOTE]
+> If you specify a hashtable for **SessionOption**, PowerShell converts the hashtable into a
+> **System.Management.Autiomation.Remoting.PSSessionOption** object. The values for keys specified
+> in the hashtable are cast to the matching property of the object. This behaves differently from
+> calling `New-PSSessionOption`. For example, the **System.TimeSpan** values for the timeout
+> properties, like **IdleTimeout**, convert an integer value into ticks instead of milliseconds.
+> For more information on the **PSSessionOption** object and its properties, see
+> [PSSessionOption](https://learn.microsoft.com/dotnet/api/system.management.automation.remoting.pssessionoption)
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Core/Invoke-Command.md:1436]:
```suggestion
> [PSSessionOption](/dotnet/api/system.management.automation.remoting.pssessionoption)
```

File: reference/7.3/Microsoft.PowerShell.Core/Invoke-Command.md [hunk 4]
```diff
@@ -1416,6 +1463,15 @@ Specifies advanced options for the session. Enter a **SessionOption** object, su
...
 names and the values are session option values.
 
+> [!NOTE]
+> If you specify a hashtable for **SessionOption**, PowerShell converts the hashtable into a
+> **System.Management.Autiomation.Remoting.PSSessionOption** object. The values for keys specified
+> in the hashtable are cast to the matching property of the object. This behaves differently from
+> calling `New-PSSessionOption`. For example, the **System.TimeSpan** values for the timeout
+> properties, like **IdleTimeout**, convert an integer value into ticks instead of milliseconds.
+> For more information on the **PSSessionOption** object and its properties, see
+> [PSSessionOption](https://learn.microsoft.com/dotnet/api/system.management.automation.remoting.pssessionoption)
```
Comment (sdwheeler) [inline reference/7.3/Microsoft.PowerShell.Core/Invoke-Command.md:1473]:
```suggestion
> [PSSessionOption](/dotnet/api/system.management.automation.remoting.pssessionoption)
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]
Fix the site URLs

**Review by sdwheeler — APPROVED** [review]

File: reference/5.1/Microsoft.PowerShell.Core/Invoke-Command.md (hunk 1 above)
Comment (michaeltlombardi) [reply reference/5.1/Microsoft.PowerShell.Core/Invoke-Command.md]: oh, good catch, tyvm

**Pull request merged**